server/src/main/java/org/opensearch/common/settings/ClusterSettings.java

```diff
@@ -760,7 +761,14 @@ public void apply(Settings value, Settings current, Settings previous) {
                 // Composite index settings
                 CompositeIndexSettings.STAR_TREE_INDEX_ENABLED_SETTING,
 
-                SystemTemplatesService.SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_ENABLED
+                SystemTemplatesService.SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_ENABLED,
+
+                // QueryGroup settings
+                QueryGroupServiceSettings.MAX_QUERY_GROUP_COUNT,
+                QueryGroupServiceSettings.NODE_LEVEL_CPU_REJECTION_THRESHOLD,
+                QueryGroupServiceSettings.NODE_LEVEL_CPU_CANCELLATION_THRESHOLD,
+                QueryGroupServiceSettings.NODE_LEVEL_MEMORY_REJECTION_THRESHOLD,
+                QueryGroupServiceSettings.NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD
```

Comment:
This should be separate PR for the settings IMO. Does not have anything to do with the CRUD API